### PR TITLE
DOC: fix a mistake in the docstring of vector_norm

### PR DESCRIPTION
**Pull request by yuanx749**
`ord` in `vector_norm` can be any integer. Also added an example for `ord=0`.
Refer to the table under Notes in [numpy.linalg.norm](https://numpy.org/devdocs/reference/generated/numpy.linalg.norm.html#numpy.linalg.norm).

### CODE REVIEW

File: numpy/linalg/_linalg.py [hunk 1]
```diff
@@ -2587,7 +2587,7 @@ def norm(x, ord=None, axis=None, keepdims=False):
         Input array.  If `axis` is None, `x` must be 1-D or 2-D, unless `ord`
         is None. If both `axis` and `ord` are None, the 2-norm of
         ``x.ravel`` will be returned.
-    ord : {non-zero int, inf, -inf, 'fro', 'nuc'}, optional
+    ord : {int, inf, -inf, 'fro', 'nuc'}, optional
```
Comment (mdhaber) [inline numpy/linalg/_linalg.py:2590]: This works for vectors, but have you tested with matrices? 
```python3
import numpy as np
x = np.ones((3, 3))
np.linalg.norm(x, ord=0)
# ValueError: Invalid norm order for matrices.
```

Comment (mdhaber) [reply numpy/linalg/_linalg.py:2590]: Then again, I suppose it's better to say any int here and clarify later, which the table does. Never mind.

Comment (mdhaber) [reply numpy/linalg/_linalg.py:2590]:
```suggestion
    ord : {int, float, inf, -inf, 'fro', 'nuc'}, optional
```

**Review by mdhaber — CHANGES_REQUESTED** [review]
Thanks @yuanx749. I'd suggest some slight additions, then I can merge this.

**Comment by yuanx749** [comment]
Thanks for the review @mdhaber . True, any float is ok.

**Review by mdhaber — APPROVED** [review]
LGTM. Thanks @yuanx749!

**Pull request merged**